buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/DialectGenerate.java

```diff
@@ -74,8 +74,10 @@
    * @param fileText The contents of the file to process
    * @param extractedData The object to which the parsed functions and token assignments
    *             will be added to
+   * @param filePath The path of the file being processed
    */
-  public void processFile(String fileText, ExtractedData extractedData) {
+  public void processFile(String fileText, ExtractedData extractedData,
+      String filePath) {
```

Comment:
Will there be other places that might call processFile without filePath that would still be a legitimate call?

Reply:
Not at the moment but I can see it possibly happening in the future. I'll make this parameter optional by having a null check before the annotation gets added.